lib/src/revset.rs

```diff
@@ -429,13 +429,16 @@ impl RevsetExpression {
     }
 
     /// Ancestors of `self`, including `self`.
-    pub fn ancestors(self: &Rc<RevsetExpression>) -> Rc<RevsetExpression> {
+    pub fn ancestors(self: &Rc<RevsetExpression>, generation: Range<u64>) -> Rc<RevsetExpression> {
         Rc::new(RevsetExpression::Ancestors {
             heads: self.clone(),
-            generation: GENERATION_RANGE_FULL,
+            generation,
             is_legacy: false,
         })
     }
+    pub fn all_ancestors(self: &Rc<RevsetExpression>) -> Rc<RevsetExpression> {
```

Comment:
Actually, `ancestors_at()` was introduced in #1200, but with a single generation as argument, so you'd have to either modify that function to take a range or add another function (I'd probably do the former).

Reply:
Ah, I just got in before `ancestors_at` existed! I'll do the latter.

Reply:
Actually afaict `ancestors_at` is for getting a single ancestor `generation` steps back, and the name refers to that? I think it would make more sense to add a new function, say `ancestors_range`. Unless you were meaning to suggest  renaming the newly range-taking `ancestors_at` to reflect its new range-taking status.

Reply:
I added a new `ancestors_range`, kept `ancestors_at`, and made `ancestors` and `ancestors_at` in terms of `ancestors_range`.